lua/orgmode/objects/priority_state.lua

```diff
@@ -0,0 +1,116 @@
+local config = require('orgmode.config')
+local utils = require('orgmode.utils')
+
+---@class PriorityState
+local PriorityState = {}
+
+---@param priority string
+function PriorityState:new(priority)
+  local o = {}
+
+  o.priority = priority
+  o.high_priority = config.org_priority_highest
+  o.low_priority = config.org_priority_lowest
+
+  if not priority or priority == '' then
+    o.priority = config.org_priority_default
+  end
+
+  setmetatable(o, self)
+  self.__index = self
+
+  return o
+end
+
+---@return string
+function PriorityState:prompt_user()
+  local choice = utils.choose(self:_get_enumerated())
+  if not choice then
+    return
+  end
+
+  return choice.choice_value
+end
+
+function PriorityState:is_alphabetical()
+  return string.find(self.high_priority, '%a')
+end
+
+---@return number
+function PriorityState:get_sort_value()
+  local priority = self.priority
+  if self:is_alphabetical() then
```

Comment:
You can actually remove the check for `alphabetical` and always convert to byte and back. `string.byte(1)` is for example 49, 2 is 50, and so on. Theoretically someone can mix both numbers and letters if they want, if it's according to this byte value.

Reply:
There is mention of ascii value here https://emacs.stackexchange.com/a/2741, which is taken from the orgmode source.

Reply:
Not sure this will work if users use multi-byte values:

```lua
print(string.char('a'))  -- 97
print(string.byte('aa'))  -- 97
print(string.byte('a7'))  -- 97
```

Reply:
I can't really figure out how emacs is handling these. From the same link I attached above, you can see that someone uses this:
```elisp
(setq org-highest-priority ?U)
(setq org-default-priority ?i)
(setq org-lowest-priority ?o)
```
Since I wasn't sure what `?` means, I googled it, and it turns out that `?` [converts the value to ascii](https://stackoverflow.com/a/19867243/1474465), so they are doing something with ascii here.

Generally it seems that emacs orgmode itself is not handling these too well, according to few of these SO questions.

I think your solution of converting both things to string and negating it will do the trick. This will reduce complexity by removing the check if it's alphabetical, and should work in most of cases.

Reply:
Did a bit more poking at this, and as mentioned [here](https://emacs.stackexchange.com/questions/29935/customize-org-priorities-with-strings):

>Emacs converts the priority letters or numbers into character numbers for purposes of sorting, etc. -- e.g., (string-to-char "A") is 65.

which makes sense why it is noted [here](https://orgmode.org/manual/Priorities.html) that all numeric priorities:

>must all be strictly inferior to 65

---

I took some time to get _Emacs_ setup to compare feature parity, and the way we are choosing priorities here is actually a bit off as well. There are two things that need to be changed:

1. The prompt should be more just like a vanilla `vim.ui.input` (or just `input`) that allows the user to type in the priority and hit `<CR>`.
2. When cycling priorities, the priority should be removed if the user goes above/below the highest priority

Going to change these two things, make sure the sorting function coincides with this and then go from there.